lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -27,6 +27,107 @@ using namespace mlir;
 using namespace mlir::torch;
 using namespace mlir::torch::Torch;
 
+// Checks the validity of pooling parameters and stores them in the respective
+// vector.
+template <typename OpTy>
+static LogicalResult
+checkAndGetPoolingParameters(OpTy op, ConversionPatternRewriter &rewriter,
+                             SmallVector<int64_t, 2> &kernelSizeInts,
+                             SmallVector<int64_t, 2> &strideInts,
+                             SmallVector<int64_t, 2> &paddingInts,
+                             SmallVector<int64_t, 2> &dilationInts) {
+  // Pattern match against the op's original operands, because otherwise we
+  // will get the lowered version of the operands which is harder to pattern
+  // match.
+  if (!matchPattern(op.kernel_size(), m_TorchConstantIntList(kernelSizeInts)))
+    return rewriter.notifyMatchFailure(op, "only support kernel size ints");
+  if (!matchPattern(op.stride(), m_TorchConstantIntList(strideInts)))
+    return rewriter.notifyMatchFailure(op, "only support constant int strides");
+  if (!matchPattern(op.padding(), m_TorchConstantIntList(paddingInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int paddings");
+  if (!matchPattern(op.dilation(), m_TorchConstantIntList(dilationInts)))
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant int dilations");
+  bool ceilMode;
+  if (matchPattern(op.ceil_mode(), m_TorchConstantBool(&ceilMode))) {
+    if (ceilMode) {
+      return rewriter.notifyMatchFailure(op,
+                                         "only ceil_mode false is supported");
+    }
+  } else
+    return rewriter.notifyMatchFailure(op,
+                                       "only support constant bool ceil_mode");
+
+  return success();
+}
+
+// Computes maxpool2d for AtenMaxPool2dOp and AtenMaxPool2dWithIndicesOp.
+template <typename OpTy>
+static LogicalResult
+computeMaxPool2d(OpTy op, ConversionPatternRewriter &rewriter, Value self,
+                 SmallVector<int64_t, 2> kernelSizeInts,
+                 SmallVector<int64_t, 2> strideInts,
+                 SmallVector<int64_t, 2> paddingInts,
+                 SmallVector<int64_t, 2> dilationInts, Value &result) {
+  Location loc = op.getLoc();
+  Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+  if (!elementType.isa<mlir::FloatType>())
+    return op.emitError("unimplemented: non-floating point type");
+
+  SmallVector<int64_t, 4> paddingIncludingNC = {0, 0};
+  paddingIncludingNC.insert(paddingIncludingNC.end(), paddingInts.begin(),
+                            paddingInts.end());
+  Value paddedInput =
+      torch_to_linalg::getPaddedTensor(op, rewriter, self, paddingIncludingNC);
```

Comment:
I don't think this is the right variant of `getPaddedTensor` to use. From the comment on it,

https://github.com/llvm/torch-mlir/blob/fa0b24a73ce4e403b077c8d894fe7a12d97303bb/lib/Conversion/TorchToLinalg/Utils.cpp#L56-L58

the function pads the tensors with zeros, which will result in the wrong maximums when dealing with negative tensors.

You should use this one instead to specify the fill value:

https://github.com/llvm/torch-mlir/blob/fa0b24a73ce4e403b077c8d894fe7a12d97303bb/lib/Conversion/TorchToLinalg/Utils.cpp#L37

Reply:
@vivekkhandelwal1 can you help to change the helper name to `getZeroPaddedTensor`?